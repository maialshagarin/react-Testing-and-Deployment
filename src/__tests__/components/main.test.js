import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Main from '../../components/main.js';


describe('<Main/>', () => {
  it('exists at the start of app', () => {
    let app = shallow(<Main />);
    expect(app.find('section').exists()).toBeTruthy();
  });

  it('The state start 0', ()=> {
    let app = mount(<Main />);
    expect(app.state('number')).toEqual(0);
  });

  it('changes the state on click - ', ()=> {
    let app = mount(<Main />);
    let buttonDown = app.find('.plusClicker');
    buttonDown.simulate('click');
    expect(app.state('number')).toEqual(1);
  });
  it('changes the state on click +', ()=> {
    let app = mount(<Main />);
    let buttonUp = app.find('.minusCliker');
    buttonUp.simulate('click');
    expect(app.state('number')).toEqual(-1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});