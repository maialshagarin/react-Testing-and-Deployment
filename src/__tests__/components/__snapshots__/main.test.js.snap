// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> renders correctly 1`] = `
<div>
  <h3>
    the number is : 
    0
     
  </h3>
  <h3>
     polarity : 
  </h3>
  <section>
    <button
      className="plusClicker"
      onClick={[Function]}
    >
      Click Me To Increase
    </button>
  </section>
  <section>
    <button
      className="minusCliker"
      onClick={[Function]}
    >
      Click Me To decrease
    </button>
  </section>
</div>
`;
